src/common/dsp/effects/AudioInputEffect.cpp

```diff
@@ -0,0 +1,245 @@
+
+#include "AudioInputEffect.h"
+
+AudioInputEffect::AudioInputEffect(SurgeStorage *storage, FxStorage *fxdata, pdata *pd)
+    : Effect(storage, fxdata, pd)
+{
+    effect_slot_type slotType = getSlotType(fxdata->fxslot);
+    if (storage && (slotType == a_insert_slot || slotType == b_insert_slot))
+    {
+        int scene = slotType == a_insert_slot ? 1 : 0;
+        for (int i = 0; i < N_OUTPUTS; i++)
+            sceneDataPtr[i] = storage->scenesOutputData.getSceneData(scene, i);
+    }
+}
+void AudioInputEffect::init_ctrltypes()
+{
+    Effect::init_ctrltypes();
+
+    // -----  Audio Input
+    fxdata->p[in_audio_input_channel].set_name("Channel");
+    fxdata->p[in_audio_input_channel].set_type(ct_percent_bipolar_stereo);
+    fxdata->p[in_audio_input_channel].posy_offset = 1;
+
+    fxdata->p[in_audio_input_pan].set_name("Pan");
+    fxdata->p[in_audio_input_pan].set_type(ct_percent_bipolar_stereo);
+    fxdata->p[in_audio_input_pan].posy_offset = 1;
+
+    fxdata->p[in_audio_input_level].set_name("Level");
+    fxdata->p[in_audio_input_level].set_type(ct_decibel_attenuation);
```

Comment:
I would use `ct_decibel_attenuation_large` so that we can basically go from almost silence (-96 dB).

Reply:
Done